### DISCUSSION TITLE
Cache python deps in github actions

### DISCUSSION THREAD

**Discussion by jobayle** [xvik/gradle-use-python-plugin#120] [Q&A]
Hi, and thank you for this plugin, works like a charm! I have the following question:

### ℹ️ how can I optimise build time on github actions?

I'm using the `setup-gradle` action that provides caching [1] of subfolders of the gradle user home ( ~/.gradle/... ), but I cannot cache the python dependencies as they are fetched in folder $CWD/.gradle/python

What would you suggest I do?

Thanks,
Jon

---
[1] https://github.com/gradle/actions/tree/main/setup-gradle#caching-build-state-between-jobs

**Comment by xvik** [comment]
Hi! You just need to configure environment path somewhere else:

```groovy
python {
   envPath = '/somewhere/else'
}
````

So virtual environment would be created in /somewhere/else instead of default location.

If you want to reference user dir then you'll have to resolve it manually, like this:

```groovy
python {
   envPath = '~/.myenv'.replace('~', System.getProperty("user.home"))
}
````

And, of course, if your project is multi-module, then it would be important to override path for all modules (using python)

edit:  direct support for home user dir reference for envPath [would be added](https://github.com/xvik/gradle-use-python-plugin/commit/6c16257076da017f861bd37d7aa9e2f55ad84ebd) in the upcoming version
